src/storage/migrations.js

```diff
@@ -54,7 +63,7 @@ function dropCache(state: GlobalState): $Shape<GlobalState> {
 }
 
 // This is the inward-facing type; see later export for jsdoc.
-const migrationsInner: {| [string]: (GlobalState) => GlobalState |} = {
+const migrationsInner: {| [string]: (LessPartialState) => LessPartialState |} = {
```

Comment:
> …Ah, reading more in this PR, I think maybe I see: we're going to end this sequence of migrations, and do migrations a different way.

Yeah. This might not be the absolute last migration we do in this framework (if we land any other PRs with migrations before we land the new AsyncStorage, they'll use this and should just add tests), but it's at least nearly the last.

In the future migration framework all this code will survive, but preserved in amber rolled up into a single migration in the new framework.


> When you say "we have a dropCache at the end", you mean that the current latest migration, 37, involves a `dropCache`, right?
> 
> Is that snapshot all that's relevant, or should it be more like "we do _and will continue to_ handle migrations of `cacheKeys` data using `dropCache`"?…

Hmm, rereading, for *this* commit really what's relevant is that latter point, yeah. I must have written the "because we have a dropCache at the end" before I wrote the parent commit which removed the one exception.

Reply:
> Does the first bullet point need an update? The new type does say that the `storeKeys` properties are represented.

It does, as did the old type: `GlobalState`. It also claims that the `discardKeys` properties may be present, and this bullet point is saying that that's false. (We pass redux-persist a list of keys to read, and it's just `storeKeys` plus `cacheKeys`.)

Another lie in this type is that it claims the `storeKeys` properties are definitely present, while in reality any or all keys might be missing. That one's not actually represented in this comment, huh! I'll add it.

Reply:
> it claims the `storeKeys` properties are definitely present, while in reality any or all keys might be missing.

Small correction to this, reflected in the new revision: any or all keys *other than `migrations`* might be missing, because when that's missing we skip trying to run the migrations.

Which means that they might be missing only because of our legacy unsound storage -- in the happy case, we only store a `migrations` key if we also store all the others, with at least their initial states as supplied by the reducers.